soc/arm/microchip_mec/mec172x/reg/mec172x_regs.h

```diff
@@ -0,0 +1,2105 @@
+	PS2_0_ACT_IRQn          = 100,
+	/* reserved 101-102 */
+	ESPI_PC_IRQn            = 103,
+	ESPI_BM1_IRQn           = 104,
+	ESPI_BM2_IRQn           = 105,
+	ESPI_LTR_IRQn           = 106,
+	ESPI_OOB_UP_IRQn        = 107,
+	ESPI_OOB_DN_IRQn        = 108,
+	ESPI_FLASH_IRQn         = 109,
+	ESPI_RESET_IRQn         = 110,
+	RTMR_IRQn               = 111,
+	HTMR_0_IRQn             = 112,
+	HTMR_1_IRQn             = 113,
+	WK_IRQn                 = 114,
+	WKSUB_IRQn              = 115,
+	WKSEC_IRQn              = 116,
+	WKSUBSEC_IRQn           = 117,
+	WKSYSPWR_IRQn           = 118,
+	RTC_IRQn                = 119,
+	RTC_ALARM_IRQn          = 120,
+	VCI_OVRD_IN_IRQn        = 121,
+	VCI_IN0_IRQn            = 122,
+	VCI_IN1_IRQn            = 123,
+	VCI_IN2_IRQn            = 124,
+	VCI_IN3_IRQn            = 125,
+	VCI_IN4_IRQn            = 126,
+	/* reserved 127-128 */
+	PS2_0A_WAKE_IRQn        = 129,
+	PS2_0B_WAKE_IRQn        = 130,
+	/* reserved 131-134 */
+	KEYSCAN_IRQn            = 135,
+	B16TMR_0_IRQn           = 136,
+	B16TMR_1_IRQn           = 137,
+	B16TMR_2_IRQn           = 138,
+	B16TMR_3_IRQn           = 139,
+	B32TMR_0_IRQn           = 140,
+	B32TMR_1_IRQn           = 141,
+	CTMR_0_IRQn             = 142,
+	CTMR_1_IRQn             = 143,
+	CTMR_2_IRQn             = 144,
+	CTMR_3_IRQn             = 145,
+	CCT_IRQn                = 146,
+	CCT_CAP0_IRQn           = 147,
+	CCT_CAP1_IRQn           = 148,
+	CCT_CAP2_IRQn           = 149,
+	CCT_CAP3_IRQn           = 150,
+	CCT_CAP4_IRQn           = 151,
+	CCT_CAP5_IRQn           = 152,
+	CCT_CMP0_IRQn           = 153,
+	CCT_CMP1_IRQn           = 154,
+	EEPROMC_IRQn            = 155,
+	ESPI_VWIRE_IRQn         = 156,
+	/* reserved 157 */
+	I2C_SMB_4_IRQn          = 158,
+	TACH_3_IRQn             = 159,
+	/* reserved 160-165 */
+	SAF_DONE_IRQn           = 166,
+	SAF_ERR_IRQn            = 167,
+	/* reserved 168 */
+	SAF_CACHE_IRQn          = 169,
+	/* reserved 170 */
+	WDT_0_IRQn              = 171,
+	GLUE_IRQn               = 172,
+	OTP_RDY_IRQn            = 173,
+	CLK32K_MON_IRQn         = 174,
+	ACPI_EC0_IRQn           = 175,
+	ACPI_EC1_IRQn           = 176,
+	ACPI_EC2_IRQn           = 177,
+	ACPI_EC3_IRQn           = 178,
+	ACPI_EC4_IRQn           = 179,
+	ACPI_PM1_IRQn           = 180,
+	MAX_IRQn
+};
+
+/*
+ * ARM CMSIS requires typedef IRQn_Type or it generates an unknown type name
+ * compiler error. ;<(
+ * We can't create new typedefs in zephyr code, so hack with a macro.
+ * NOTE: C macros and typedef's are not equivalent! Some esoteric use cases
+ * may not generate expected code!
+ */
+#define IRQn_Type enum IRQn
+
+#define __CM4_REV 0x0201	/*!< Core Revision r2p1 */
+
+#define __MPU_PRESENT		1
+#define __VTOR_PRESENT		1
+#define __NVIC_PRIO_BITS	3
+#define __Vendor_SysTickConfig	0
+#define __FPU_PRESENT		1
+#define __FPU_DP		0
+#define __ICACHE_PRESENT	0
+#define __DCACHE_PRESENT	0
+#define __DTCM_PRESENT		0
+
+/* Requires: modules/hal/cmsis */
+#include <core_cm4.h>
+
```

Comment:
I'd ask to remove any register, defines, etc that would belong with a driver instead.  For example, wdt_regs would better be in a header with the wdt driver.

Reply:
OK. I will keep the register structures that don't have drivers and remove the others. Future driver PR's will add the register structures to the driver.

Reply:
it is very common to have all registers of an soc come in one header, if we start asking for this type of split, it will be very difficult to maintain such headers across the tree, especially if such headers are traditionally maintained in some type of HAL and we are asking for moving this into Zephyr.
 
IMO this should be fine, given that every other SOC using hal modules is able to do that right now already, look at STM32 for example, but many others...

Reply:
> it is very common to have all registers of an soc come in one header, if we start asking for this type of split, it will be very difficult to maintain such headers across the tree, especially if such headers are traditionally maintained in some type of HAL and we are asking for moving this into Zephyr.
> 
> IMO this should be fine, given that every other SOC using hal modules is able to do that right now already, look at STM32 for example, but many others...

Its one thing if this is in a HAL and managed outside, but if its in Zephyr and the aim here seems to be native drivers, than I think its best that such headers/structs/defines be with the driver.  Thus you aren't questioning which struct called wdt_regs goes with the driver, its clear.  Otherwise we'd need to rename/namespace such structs with the SoC family or something.

Reply:
For now I will minimize the number of structures in the soc header.

Reply:
Removed include of core_cm4.h from my header.  The zephyr cmsis.h under aarch32 pulls in core_cm4.h if one doesn't define some of the ARM CMSIS symbols.